node/src/options.rs

```diff
@@ -92,9 +92,21 @@ pub struct RunOptions {
     #[clap(long)]
     pub p2p_outbound_connection_timeout: Option<u64>,
 
-    /// Address to bind RPC to.
+    /// Address to bind http RPC to.
     #[clap(long, value_name = "ADDR")]
-    pub rpc_addr: Option<SocketAddr>,
+    pub http_rpc_addr: Option<SocketAddr>,
+
+    /// Enable/Disable http RPC.
+    #[clap(long)]
+    pub http_rpc_enabled: Option<bool>,
+
+    /// Address to bind websocket RPC to.
+    #[clap(long, value_name = "ADDR")]
+    pub ws_rpc_addr: Option<SocketAddr>,
+
+    /// Enable/Disable websocket RPC.
+    #[clap(long)]
+    pub ws_rpc_enabled: Option<bool>,
```

Comment:
Here's not very well thought out suggestion. Maybe the combination of the `SocketAddr` and `bool` could be handled by having a special value for disabling the address, like `--ws-rpc-addr=none` or `--ws-rpc-addr=disable`?

Using a custom type like
```rust
enum SocketAddrConfig { Disabled, Enabled(SorcketAddr) }
impl FormStr for SocketAddrConfig { /* handle "none" here */ }
```
the `ws_rpc_addr` and `http_rpc_addr` could be `Option<SocketAddrConfig>`.

I don't have a problem with the current approach for now, the configuration handling needs a bit more thought anyway so can be done separately.

Reply:
I don't want to deal with this until the configuration issue is solved. It's basically beating a dead horse at this point.

Reply:
To be clear, I'm talking about #446 . This has wasted enough time in this PR.